src/js/common/helpers/login-helpers.js

```diff
@@ -39,6 +39,8 @@ export function getUserData(dispatch) {
       dispatch(updateProfileField('status', json.data.attributes.va_profile.status));
       dispatch(updateProfileField('services', json.data.attributes.services));
       dispatch(updateLoggedInStatus(true));
+    } else {
+      dispatch(profileLoadingFinished());
```

Comment:
Yeah, I think I like the "do this thing" approach over "make these changes." When we dispatch actions to "do this thing," if we need to change what "this thing" does, it's all consolidated in the reducer instead of wherever the actions were `dispatch`ed. Also, like @jbalboni said, it makes the history more meaningful.

I think we'll also get minute performance gains by not waiting around for the next redux cycle to catch the remaining dispatched actions (however that works under the hood).